staging/src/k8s.io/client-go/examples/read-yaml-configuration-deployment/main.go

```diff
@@ -0,0 +1,214 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Note: the example only works with the code within the same release/branch.
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	appsv1beta1 "k8s.io/api/apps/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/yaml"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+func main() {
+
+	// Flag "--kubeconfig <kubeconfig-file>"
+	//
+	// Specify another kubeconfig file than "~/.kube/config"
+	//
+	var kubeconfig *string
+	if home := homedir.HomeDir(); home != "" {
+		kubeconfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+	} else {
+		kubeconfig = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+	}
+
+	// Flag "-f <yaml-file>"
+	//
+	// Specify another YAML configuration file than "./configuration.yaml"
+	//
+	var yamlfilename *string
+	yamlfilename = flag.String("f", "configuration.yaml", "(optional) absolute path to the YAML configuration file")
+
+	// Parse commandline parameters
+	flag.Parse()
+
+	// Get filename
+	yamlFilepath, err := filepath.Abs(*yamlfilename)
+
+	// Get reader from file opening
+	reader, err := os.Open(yamlFilepath)
+	if err != nil {
+		panic(err)
+	}
+
+	// Split YAML into chunks or k8s resources, respectively
+	yamlDecoder := yaml.NewDocumentDecoder(ioutil.NopCloser(reader))
+
+	// Create decoding function used for YAML to JSON decoding
+	decode := scheme.Codecs.UniversalDeserializer().Decode
+
+	// Read first resource - expecting deployment with size < 2048
+	yamlDeployment := make([]byte, 2048)
+	_, err = yamlDecoder.Read(yamlDeployment)
+	if err != nil {
+		panic(err)
+	}
+
+	// Trim unnecessary trailing 0x0 signs which are not accepted
+	TrimmedYamlDeployment := strings.TrimRight(string(yamlDeployment), string(byte(0)))
+
+	// Decode deployment resource from YAML to JSON
+	jsonDeployment, _, err := decode([]byte(TrimmedYamlDeployment), nil, nil)
+	if err != nil {
+		panic(err)
+	}
+
+	// Check "kind: deployment"
+	if jsonDeployment.GetObjectKind().GroupVersionKind().Kind != "Deployment" {
+		panic(fmt.Sprintf("\nError: Not \"Kind: Deployment\"\n%#v\n", jsonDeployment))
+	}
+
+	// Marshall JSON deployment
+	d, err := json.Marshal(&jsonDeployment)
+	if err != nil {
+		panic(err)
+	}
+
+	// Unmarshall JSON into deployment struct
+	var deployment appsv1beta1.Deployment
+	err = json.Unmarshal(d, &deployment)
```

Comment:
wasn't jsonDeployment already of this type?

Reply:
I couldn't find out how to create the 'container' for "groupVersionKind' without knowing beforehand.

Reply:
The object created by the codec should be the Deployment. You can type cast it.